### PR TITLE
Refine api 1

### PR DESCRIPTION
**Pull request by nzcodarnoc**
Make dockerfile do all the build so we can leverage it's cache.
Wire up rhubarb to an internal API
Expose rhubarb over express

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,45 +5,53 @@ Exposing DanielSWolf / rhubarb-lip-sync as a service
...
 
 Build and watch files for changes, and start the express server with hot swapping enabled
 
-```
+```bash
 npm run build & npm start
 ```
 
-Visit http://localhost:3000/ and you should see the Rhubarb version installed.
+Visit <http://localhost:3000/> and you should see the Rhubarb version installed.
```
Comment (mermop) [inline README.md:57]: I did not know this was a shorthand for a self link in markdown. TIL!

File: src/api/get-shapes.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    audioFileName: String(audioFileName),
+    shapesFileName: audioFileName ? shapesFileName : String(false),
+    shapesID: shapesID(shapesFileName),
+    valid: audioFileName ? true : false
+  };
+  if (!fileNames.valid) {
+    return fileNames;
+  }
+  const cmd = `${rhubarbCmd} -r phonetic -f json -o ${fileNames.shapesFileName} ${fileNames.audioFileName}`;
+  console.info(`cmd: ${cmd}`);
```
Comment (mermop) [inline src/api/get-shapes.ts:52]: Is this a standard way to log for a Node app?

File: src/api/get-shapes.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
+import getShapes from "./get-shapes";
+
+test("Passing a URL (of a WAV) returns the location of the mouth shape analysis and location of the downloaded WAV file.", async () => {
```
Comment (mermop) [inline src/api/get-shapes.spec.ts:4]: ✨ Love to see tests

File: src/api/get-shapes.spec.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (mermop) [inline src/api/get-shapes.spec.ts:1]: Is there a reason why there's a blank line here?

File: src/api/get-shapes.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  audioFileName: string;
+  shapesID: string;
+  valid: boolean;
+}
+
+const shapesID = (path: string): string => {
+  const parts: Array<string> = path.split("/");
+  const fileName: string = parts[parts.length - 1];
+  const things: Array<string> = fileName.split(".");
+  return things[0];
```
Comment (mermop) [inline src/api/get-shapes.ts:18]: Is `things` a technical term here..

File: src/index.ts [hunk 6]
```diff
@@ -29,9 +27,25 @@ app.use(helmet());
 app.use(cors());
 app.use(express.json());
 app.get("/", function(_req, res) {
-  res.send(rhubarbVersion);
+  res.send("OK. " + rhubarbVersion);
 });
 
+// TODO, tidy up api documentation
```
Comment (mermop) [inline src/index.ts:33]: I'd recommend deleting these commented out things, if you need them again you can always check back in the commit history

File: src/utils.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+export const PORT = parseInt(String(process.env.PORT), 10);
```
Comment (mermop) [inline src/utils.ts:3]: Is it worth making this fail more explicitly if PORT isn't set, or set a sensible default?

**Comment by nzcodarnoc** [comment]
It's a quick and dirty way to log. The best practice is to use a library like "Winston" to better control where the output goes.

**Comment by nzcodarnoc** [comment]
Tests give me life (true story)

**Comment by nzcodarnoc** [comment]
Nope

**Comment by nzcodarnoc** [comment]
I am so open to a better variable name here. Ideas?

**Comment by nzcodarnoc** [comment]
Sure thing

**Comment by nzcodarnoc** [comment]
For sure

File: src/api/get-shapes.ts (hunk 5 above)
Comment (mermop) [reply src/api/get-shapes.ts:18]: `fileNameSections`?

File: src/api/get-shapes.ts (hunk 2 above)
Comment (nzcodarnoc) [reply src/api/get-shapes.ts:52]: Yep. It will be captured in any logs on the place it is running.

File: src/api/get-shapes.spec.ts (hunk 3 above)
Comment (nzcodarnoc) [reply src/api/get-shapes.spec.ts:3]: ✨

File: src/api/get-shapes.ts (hunk 5 above)
Comment (nzcodarnoc) [reply src/api/get-shapes.ts]: Thank you! I've renamed it.

File: src/index.ts (hunk 6 above)
Comment (nzcodarnoc) [reply src/index.ts]: Done

File: src/utils.ts (hunk 7 above)
Comment (nzcodarnoc) [reply src/utils.ts:10]: I've added the validation library for env vars in nodejs

**Pull request merged**